### PR TITLE
core concepts

### PR DESCRIPTION
**Pull request by dsuxo**

### CODE REVIEW

File: src/components/counter/counter.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import "./counter.css";
+
+class Counter extends React.Component{
+    state = {
+        counterValue : 0
+    };
+
+incrementCounter(){
+    let initialValue = this.state.counterValue;
```
Comment (nikita-muratov-epam) [inline src/components/counter/counter.js]: I would advise to use the second form of setState:
`this.setState((state) => ({
  counteValuer: state.counterValue + 1,
}));`

File: src/components/counter/counter.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    this.setState({
+        counterValue: initialValue - 1
+    });
+}
+render(){
+    return (
+        <div className = "regularDiv">
+            <h1>Counter: {this.state.counterValue}</h1>
+            <button onClick={this.incrementCounter.bind(this)}>increment</button>
```
Comment (nikita-muratov-epam) [inline src/components/counter/counter.js]: I would advise to bind the methods in the constructor or use other methods of handling this.
Please review this page: [https://reactjs.org/docs/handling-events.html](https://reactjs.org/docs/handling-events.html)

**Review by nikita-muratov-epam — CHANGES_REQUESTED** [review]
Hello,
Overall everything looks good!
Great job!
I've left a couple of comments - please make changes according to them and I'll approve the task.
Thank you.

File: src/components/counter/counter.js (hunk 1 above)
Comment (dsuxo) [reply src/components/counter/counter.js]: thanks changed

File: src/components/counter/counter.js (hunk 2 above)
Comment (dsuxo) [reply src/components/counter/counter.js]: thanks changed

**Review by nikita-muratov-epam — APPROVED** [review]

**Pull request merged**